frontend/components/customizer/query-string-builder.js

```diff
@@ -36,255 +36,246 @@ function getBadgeOption(name) {
   return supportedBadgeOptions.find(opt => opt.name === name)
 }
 
+function getQueryString({ queryParams, badgeOptions }) {
+  const outQuery = {}
+  let isComplete = true
+
+  Object.entries(queryParams).forEach(([name, value]) => {
+    // As above, there are two types of supported params: strings and
+    // booleans.
+    const isStringParam = typeof value === 'string'
+    if (isStringParam) {
+      if (value) {
+        outQuery[name] = value
+      } else {
+        // Skip empty params.
+        isComplete = false
+      }
+    } else {
+      // Generate empty query params for boolean parameters by translating
+      // `{ compact_message: true }` to `?compact_message`. When values are
+      // false, skip the param.
+      if (value) {
+        outQuery[name] = null
+      }
+    }
+  })
+
+  Object.entries(badgeOptions).forEach(([name, value]) => {
+    const { shieldsDefaultValue } = getBadgeOption(name)
+    if (value && value !== shieldsDefaultValue) {
+      outQuery[name] = value
+    }
+  })
+
+  const queryString = stringifyQueryString(outQuery)
+
+  return { queryString, isComplete }
+}
+
+function ServiceQueryParam({
+  name,
+  value,
+  exampleValue,
+  isStringParam,
+  stringParamCount,
+  handleServiceQueryParamChange,
+}) {
+  return (
+    <tr>
+      <td>
+        <QueryParamLabel htmlFor={name}>
+          {humanizeString(name).toLowerCase()}
+        </QueryParamLabel>
+      </td>
+      <td>
+        {isStringParam && (
+          <QueryParamCaption>
+            {stringParamCount === 0 ? `e.g. ${exampleValue}` : exampleValue}
+          </QueryParamCaption>
+        )}
+      </td>
+      <td>
+        {isStringParam ? (
+          <QueryParamInput
+            name={name}
+            onChange={handleServiceQueryParamChange}
+            type="text"
+            value={value}
+            {...noAutocorrect}
+          />
+        ) : (
+          <input
+            checked={value}
+            name={name}
+            onChange={handleServiceQueryParamChange}
+            type="checkbox"
+          />
+        )}
+      </td>
+    </tr>
+  )
+}
+
+function BadgeOptionInput({ name, value, handleBadgeOptionChange }) {
+  if (name === 'style') {
+    return (
+      <select name="style" onChange={handleBadgeOptionChange} value={value}>
+        {advertisedStyles.map(style => (
+          <option key={style} value={style}>
+            {style}
+          </option>
+        ))}
+      </select>
+    )
+  } else {
+    return (
+      <QueryParamInput
+        name={name}
+        onChange={handleBadgeOptionChange}
+        type="text"
+        value={value}
```

Comment:
What's the difference between `checked={value}` (before) and this?

Reply:
`checked` is used for the state of a check box. This is a text input, so I think it should always have been using `value` instead of `checked`.